### ISSUE TITLE
Implement global structure metrics (Category D)

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#199]
## Summary

Implement the 4 global structure metrics (Category D) that measure how well
UMAP embeddings preserve large-scale distance relationships and ordering
beyond local neighborhoods.

## Context

- Plan reference: `docs/merfish-umap-benchmark-plan.md` Batch 4, Issue 4.3
- Metric definitions: `docs/metrics/global-structure.md`
- Category B (spatial, #195) and Category C (cluster, #197) are in progress
- This is independent of #195 and #197 — can be implemented in parallel

### Module: `tests/visual_eval/global_metrics.py`

Pure numpy/scipy/sklearn module — no scanpy or anndata dependency.

#### Metric 1: Random Triplet Accuracy

```python
def random_triplet_accuracy(
    X_high: np.ndarray,          # (n, d) high-dim (PCA features)
    X_low: np.ndarray,           # (n, 2) UMAP embedding
    n_triplets: int = 50000,
    seed: int = 42,
) -> float:
```
For random triplets (i, j, k): check if `d(i,j) < d(i,k)` ordering is
preserved between high-dim and embedding. Subsample points to max 5000
before computing distance matrices. Range [0.5, 1.0] (0.5 = random baseline).

#### Metric 2: Shepard Diagram Correlation

```python
def shepard_correlation(
    X_high: np.ndarray,
    X_low: np.ndarray,
    sample_n: int = 2000,
    seed: int = 42,
) -> dict:
```
Pearson and Spearman correlation of pairwise distances (high-dim vs
embedding). Subsample to `sample_n` points. Returns
`{"pearson": float, "spearman": float}`.

#### Metric 3: Centroid Distance Correlation

```python
def centroid_distance_correlation(
    X_high: np.ndarray,
    X_low: np.ndarray,
    labels: np.ndarray,
) -> float:
```
Spearman correlation of inter-cluster centroid distances in high-dim vs
embedding. Cheap global metric — O(c²) where c = number of clusters.

#### Metric 4: kNN Preservation Rate

```python
def knn_preservation(
    X_high: np.ndarray,
    X_low: np.ndarray,
    k: int = 15,
) -> float:
```
Mean fraction of k-nearest neighbors preserved from high-dim to embedding.
Range [0, 1], higher = better local preservation.

#### Convenience Function

```python
def compute_global_metrics(
    X_high: np.ndarray,
    X_low: np.ndarray,
    labels: np.ndarray,
    k: int = 15,
) -> dict:
```
Runs all 4 metrics and returns a dict with keys: `triplet_accuracy`,
`shepard_pearson`, `shepard_spearman`, `centroid_dist_corr`, `knn_preservation`.

## Dependencies

- `numpy`, `scipy.spatial.distance.pdist`, `scipy.stats.spearmanr`, `scipy.stats.pearsonr`
- `sklearn.neighbors.NearestNeighbors`

## Testing

- Smoke test: PCA 2D projection of blobs should score higher on all metrics
  than a random permutation of coordinates
- Triplet accuracy on identity mapping (X_low = X_high[:, :2]) should be ~1.0
### API — Function Signatures

- **REQ-API-001:** Each metric function must accept numpy arrays as input — no AnnData or scanpy objects.
- **REQ-API-002:** Each metric function must return a float (or dict of floats for Shepard).
- **REQ-API-003:** A `compute_global_metrics()` convenience function must exist that runs all 4 metrics and returns a dict.
- **REQ-API-004:** The module must have zero dependency on scanpy, anndata, or polars.

### TRIP — Random Triplet Accuracy

- **REQ-TRIP-001:** Triplet accuracy must subsample points to at most 5000 before computing distance matrices.
- **REQ-TRIP-002:** Triplet accuracy must use `seed=42` for reproducible triplet selection.
- **REQ-TRIP-003:** Triplets where any two indices are equal must be excluded.

### SHEP — Shepard Diagram Correlation

- **REQ-SHEP-001:** Shepard correlation must compute both Pearson and Spearman correlations of pairwise distances.
- **REQ-SHEP-002:** Shepard correlation must subsample to at most `sample_n` points (default 2000).

### CENT — Centroid Distance Correlation

- **REQ-CENT-001:** Centroid distance correlation must compute per-label centroids in both spaces and correlate inter-centroid `pdist` vectors via Spearman.

### KNN — kNN Preservation Rate

- **REQ-KNN-001:** kNN preservation must compute the mean fraction of k-nearest neighbors in high-dim that are also k-nearest in the embedding.
- **REQ-KNN-002:** kNN preservation must default to `k=15` matching the UMAP `n_neighbors` parameter.

### TEST — Smoke Tests

- **REQ-TEST-001:** A smoke test must verify that a PCA 2D projection scores higher on all metrics than a random coordinate permutation.
- **REQ-TEST-002:** Triplet accuracy on an identity-like mapping must approach 1.0.
